engine/gamesys/src/gamesys/components/comp_collision_object.cpp

```diff
@@ -1216,16 +1216,16 @@ namespace dmGameSystem
         return GetGroupBitIndex((CollisionWorld*)world, group_hash);
     }
 
-    void RayCast(void* _world, const dmPhysics::RayCastRequest& request, dmPhysics::RayCastResponse& response)
+    void RayCast(void* _world, const dmPhysics::RayCastRequest& request, dmArray<dmPhysics::RayCastResponse>& results)
```

Comment:
One design issue was how to allow for an arbitrary number of results.

My first take was to use callbacks, but the scripting interface became a bit too cumbersome.
I then thought of passing a pointer to a pointer, to allow for the function to allocate a buffer, but I instead opted to use our dynamic array class.

I don't feel that any option really were "clean", but this eventually felt cleanest in the end.